zipkin-storage/cassandra3/src/main/java/zipkin/storage/cassandra3/CassandraSpanConsumer.java

```diff
@@ -103,43 +102,29 @@
       // indexing occurs by timestamp, so derive one if not present.
       Long timestamp = guessTimestamp(span);
       TraceIdUDT traceId = new TraceIdUDT(span.traceIdHigh, span.traceId);
-      boolean isServerRecvSpan = isServerRecvSpan(span);
-      futures.add(storeSpan(span, traceId, isServerRecvSpan, timestamp));
+      futures.add(storeSpan(span, traceId, timestamp));
 
       for (String serviceName : span.serviceNames()) {
-        // QueryRequest.min/maxDuration
-        if (timestamp != null) {
-          // Contract for Repository.storeTraceServiceSpanName is to store the span twice, once with
-          // the span name and another with empty string.
-          futures.add(storeTraceServiceSpanName(serviceName, span.name, timestamp, span.duration,
-              traceId));
-          if (!span.name.isEmpty()) { // If span.name == "", this would be redundant
-            futures.add(storeTraceServiceSpanName(serviceName, "", timestamp, span.duration, traceId));
-          }
-          futures.add(storeServiceSpanName(serviceName, span.name));
+        if (timestamp == null) continue; // QueryRequest.min/maxDuration needs timestamp
```

Comment:
cc @llinder fyi this was just for sanity as indent level was blinding. no logic change